### PR TITLE
Include MarshalAs attributes in the collection assigned to CustomAttributes property

### PR DESCRIPTION
**Pull request by MaximLipnin**
Fixes: #7615

### CODE REVIEW

**Comment by MaximLipnin** [comment]
@lambdageek Could you also take a look at this draft? Is it correct way to proceed?

File: mcs/class/corlib/System/MonoCustomAttrs.cs [hunk 1]
```diff
@@ -301,6 +301,35 @@ internal static IList<CustomAttributeData> GetCustomAttributesData (ICustomAttri
...
 			CustomAttributeData [] attrs = GetCustomAttributesDataInternal (obj);
+
+			object [] pseudoAttrs = System.Array.FindAll (GetPseudoCustomAttributes (obj, null), attr => attr is MarshalAsAttribute);
+			if (pseudoAttrs.Length > 0)
+			{
+				CustomAttributeData [] maAttrsData = new CustomAttributeData [pseudoAttrs.Length];
+				for (var i = 0; i < pseudoAttrs.Length; i++) {
+					MarshalAsAttribute maAttr = (MarshalAsAttribute)pseudoAttrs[i];
+					var attrValue = maAttr.Value;
+					var customAttributeData = (CustomAttributeData)Activator.CreateInstance (typeof (CustomAttributeData), true);
```
Comment (lambdageek) [inline mcs/class/corlib/System/MonoCustomAttrs.cs]: Maybe add a static method to `CustomAttributeData.CreateFromPseudoCustomAttr()` that initializes a `CustomAttributeData` instance directly with the ctorArgs and ctorInfo?  Then you wouldn't have to jump through reflection.

File: mcs/class/corlib/System/MonoCustomAttrs.cs [hunk 2]
```diff
@@ -301,6 +301,35 @@ internal static IList<CustomAttributeData> GetCustomAttributesData (ICustomAttri
 				throw new ArgumentNullException ("obj");
 
 			CustomAttributeData [] attrs = GetCustomAttributesDataInternal (obj);
+
+			object [] pseudoAttrs = System.Array.FindAll (GetPseudoCustomAttributes (obj, null), attr => attr is MarshalAsAttribute);
```
Comment (lambdageek) [inline mcs/class/corlib/System/MonoCustomAttrs.cs]: Should we do this for all pseudo custom attributes, not just `MarshalAs`?

**Comment by marek-safar** [comment]
I commented on the original issue we are missing handling for all pseudo custom attribute in this scenario and we need more light-weight way to get them (less allocation)

**Pull request closed without merging**